test/tools/d_do_test.d

```diff
@@ -426,6 +426,20 @@ string execute(ref File f, string command, bool expectpass, string result_path)
     return output;
 }
 
+/// add quotes around the whole string if it contains spaces that are not in quotes
+string quoteSpaces(string str)
+{
+    if (str.indexOf(' ') < 0)
```

Comment:
Would this smaller version work?
```D
if (str.startsWith(`"`) || str.indexOf(' ') < 0)
    return str;
return `"` ~ str ~ `"`;
```

Reply:
Not with the current default:
https://github.com/dlang/dmd/blob/edc9b35b85f769081c078df353363f9dc753a3aa/test/tools/d_do_test.d#L634

Reply:
Jesus bash is friggin weird

Reply:
If you want it to also handles spaces on posix systems, you'll probably need a different implementation for posix.  But not a big deal since there's probably alot of other stuff that wouldn't work with the DMD build if there were spaces in the path.

Reply:
> If you want it to also handles spaces on posix systems, you'll probably need a different implementation for posix. But not a big deal since there's probably alot of other stuff that wouldn't work with the DMD build if there were spaces in the path.

I also wouldn't expect spaces in filenames on posix to be battle tested. It's also quite a mess dealing with them between bash, make and every other tool.